### DISCUSSION TITLE
-out_stat option for StatAnalysis WDIR

### DISCUSSION THREAD

**Discussion by lindsayrblank** [dtcenter/METplus#1370] [Statistical Computation] [answered]
Hello folks,

I am running StatAnalysis and calculating wind direction statistics using the '-out_line_type WDIR' flag. I am unable to set the '-out_stat' flag to create a file with the output. Instead, I have to pipe it out while running the command. Is there a flag that I am unaware of that acts like '-out_stat' for WDIR?

Best,
Lindsay

**Comment by jprestop** [comment, marked as answer]
Hi @lindsayrblank.  I talked with @JohnHalleyGotway and he indicated that unfortunately, the output for the WDIR job in Stat-Analysis is NOT a true, supported STAT line type. Our support for the verification of wind direction has been choppy and piecemeal, and this is one of the pieces. We wish we could go back in time, figure out the right way to handle it from the beginning and provide consistent and intuitive functionality.

Using "-out_stat" won't have any effect. Using "-out" will write data to an ascii output file, but it can't be loaded into a database for use in METviewer.

Here is the documentation that exists for that job: [https://met.readthedocs.io/en/latest/Users_Guide/stat-analysis.html?highlight=wdir#wind-direction-statistics](https://met.readthedocs.io/en/latest/Users_Guide/stat-analysis.html?highlight=wdir#wind-direction-statistics)

Please take a look at the VCNT line type, and esp if you're still processing MPR input lines for stat-analysis. 

John put together an example for you:

```
bin/stat_analysis \
-lookin out/point_stat/point_stat_360000L_20070331_120000V_mpr.txt \
-job aggregate_stat -line_type MPR -out_line_type VCNT \
-by OBS_SID,INTERP_MTHD,INTERP_PNTS \
-column_str OBS_SID KWWR,KWVL \
-out_stat mpr_to_vcnt.txt

```
And attached is the output mpr_to_vcnt.txt file that was generated:
[mpr_to_vcnt.txt](https://github.com/dtcenter/METplus/files/7934914/mpr_to_vcnt.txt)


The input file is generated when you run "make test". But I uploaded it here as well:
[point_stat_360000L_20070331_120000V_mpr.txt](https://github.com/dtcenter/METplus/files/7934916/point_stat_360000L_20070331_120000V_mpr.txt)


For this example, he only used data for stations named KWWR and KWVL. The aggregate_stat job reads MPR lines and computes vector continuous stats output lines. It does it separately for each combo of station id and interpolation options. No very exciting computing vector stats for only 3 points... but in the past, this is exactly how he thought you had been running Stat-Analysis on MPR input lines.

I hope this helps.  Please let us know what other questions you have, or please select this answer as the answer if you no longer have any questions.

**Reply by lindsayrblank** [reply]
Hi @jprestop! Thank you for your detailed response and to @JohnHalleyGotway for the example. VCNT produces exactly what I want (the MAE of the wind direction) and since it outputs a traditional MET stat file with headers, this will make ingestion much easier. Thanks!

**Reply by jprestop** [reply]
Great, Lindsay, thanks!  I'm going to mark this discussion as being answered and "lock it" to prevent future posts. That's how we encourage users to create new discussions for new questions.  Please feel free to create a new Discussion with any other questions you have.
